src/crypto/secp156k1_provider/secp256k1_provider_impl.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "libp2p/crypto/secp256k1_provider/secp256k1_provider_impl.hpp"
+
+#include <memory>
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/obj_mac.h>
+#include "libp2p/crypto/error.hpp"
+#include "libp2p/crypto/sha/sha256.hpp"
+
+namespace libp2p::crypto::secp256k1 {
+  outcome::result<KeyPair> Secp256k1ProviderImpl::generateKeyPair() const {
+    PublicKey publicKey{};
```

Comment:
```suggestion
    PublicKey public_key{};
```
according to our style guide, local variables should be named in snake case.
sometimes this rule is not met, but better comply with it.